dpe/src/dpe_instance.rs

```diff
@@ -389,6 +389,83 @@ impl DpeInstance {
 
         hasher.finish().map_err(|_| DpeErrorCode::HashError)
     }
+
+    /// Retrieve index of the root context based on the tci_type and locality of
+    /// the context matching the inputs
+    ///
+    /// # Arguments
+    ///
+    /// * `idx` - desired index of the root context
+    /// * `locality` - desired locality of the root context
+    pub fn get_root_idx(&self, tci_type: u32, locality: u32) -> Result<usize, DpeErrorCode> {
+        // find indices of all non-inactive contexts matching tci_type and locality
+        let mut tci_nodes = self.contexts.iter().enumerate().filter(|(_, context)| {
+            context.state != ContextState::Inactive
+                && context.locality == locality
+                && context.tci.tci_type == tci_type
```

Comment:
tci_type isn't guaranteed to be unique, so I don't think we can use this for searching for a specific node. What is the use-case for this function during validation?

Reply:
This is to find the root node with tci_type "RTJM". I think the root node is the only node with this tci_type.

Reply:
On second thought, I think this function along with get_current_tci are pretty niche and would be better off on the caliptra side, so I deleted them from this PR.

Reply:
Ok, sounds good. The caller to DeriveChild can pick any value they want for the type, including "RTJM". I think another way we could find the root node is the find an Active node whose parent is 0xFF (no parent) and that there is only one such node.